hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/index/bucket/HoodieSparkConsistentBucketIndex.java

```diff
@@ -275,4 +278,46 @@ public Option<HoodieRecordLocation> getRecordLocation(HoodieKey key) {
       throw new HoodieIndexException("Failed to getBucket as hashing node has no file group");
     }
   }
+
+  /**
+   * Update default metadata file(00000000000000.hashing_meta) with the latest committed metadata file so that default file will be in sync
+   * with latest commit.
+   *
+   * @param table
+   */
+  public void updateMetadata(HoodieTable table) {
+    Map<String, Boolean> partitionVisiteddMap = new HashMap<>();
+    HoodieTimeline hoodieTimeline = table.getActiveTimeline().getCompletedReplaceTimeline();
+    hoodieTimeline.getInstants().forEach(instant -> {
+      Option<Pair<HoodieInstant, HoodieClusteringPlan>> instantPlanPair =
```

Comment:
When replacecommit comes from `INSERT OVERWRITE` action, is there any problem here?

Reply:
No i think, every clustering operation on consistent hashing index engine will create new metadata file, that will be  **<instant_time>.hashing_meta** , this particular piece of code will make **<instant_time>.hashing_meta** in sync with **00000000000000.hashing_meta** before triggering of archival.

Reply:
@rohan-uptycs, the `getCompletedReplaceTimeline()` returns the completed replacecommit, which could generate from clustering operation and insert overwrite action. No problem?

Reply:
Oh okay got it.I don't see any issue with insert overwrite action ,with insert overwrite action **ClusteringUtils.getClusteringPlan** will return empty plan for insert overwrite action. Just added to check to ignore empty plan.

Thanks for pointing out.

Reply:
@rohan-uptycs, BTW, does the replacecommit in archived timeline need to update metadata?

Reply:
@SteNicholas , using all replace commits to update metadata will not cause any issue. Its just that in some cases it will update default metadata file multiple times, until it get archived.

Ideally it should update metadata for replace commit which are going to get archived **replacecommits before oldestInstantToRetainForClustering** .

Reply:
@SteNicholas , Updated the code to update metadata for replaced commits which are eligible for archival in the current run. This will avoid un-necessary metadata update.
Please review .